v1/writable/image.go

```diff
@@ -0,0 +1,285 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package writable
```

Comment:
See my comments [here](https://github.com/google/go-containerregistry/pull/62#issuecomment-381373524) and [here](https://github.com/google/go-containerregistry/pull/62#issuecomment-381618720), but my hope is that we can elide 99% of this boilerplate through a more focused variant of this interface for appending.

Reply:
I was hoping this would be merged first :(

Reply:
No need - we updated our `writable`package

Reply:
I would say I like the concept of a `writable.Image` as you could potentially delete layers, update config information etc. in the future